### PR TITLE
Added a set of smash tests for the python plugin

### PR DESCRIPTION
**Pull request by dralley**

### CODE REVIEW

**Comment by dralley** [comment]
This is really just the basic set of tests and doesn't actually test much yet.  I'll get the others working also, but not in the first commit.

**Comment by dralley** [comment]
@kersommoura I fixed that issue but I don't understand the new error message it's providing?  That file exists.. am I missing something?

File: docs/api.rst [hunk 1]
```diff
@@ -154,6 +154,12 @@ developers, not a gospel.
     api/pulp_smash.tests.pulp3.pulpcore.api_v3.test_status
     api/pulp_smash.tests.pulp3.pulpcore.api_v3.test_workers
     api/pulp_smash.tests.pulp3.pulpcore.utils
+    api/pulp_smash.tests.pulp3.python
+    api/pulp_smash.tests.pulp3.python.api_v3
+    api/pulp_smash.tests.pulp3.python.api_v3.test_crd_publications
```
[the remark this answers is not in the document]
Comment (dralley) [reply docs/api.rst]: You're right, I was looking at the wrong branch.  This is embarrassing, lol.

File: pulp_smash/tests/pulp3/python/api_v3/utils.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        return REMOTE_DOWN_POLICY - {'background', 'on_demand'}
+    return REMOTE_DOWN_POLICY
+
+
+def gen_remote():
+    """Return a semi-random dict for use in creating an remote."""
+    return {
+        'download_policy': sample(get_remote_down_policy(), 1)[0],
+        'name': utils.uuid4(),
+        'projects': "['shelf-reader']",
```
Comment (kersommoura) [inline pulp_smash/tests/pulp3/python/api_v3/utils.py]: Is there a need to be a specific project?

If yes, please, add a constant with the project name.

Besides that, ``download_policy`` and ``sync_mode`` were just removed from the code base.

See: https://github.com/PulpQE/pulp-smash/pull/967

Please, adjust accordingly.

Comment (dralley) [reply pulp_smash/tests/pulp3/python/api_v3/utils.py]: @kersommoura  updated

**Comment by omaciel** [comment]
@dralley as per https://github.com/pulp/pulp_ansible/pull/13#issuecomment-383948271 I'd recommend that we move these tests to the `pulp_python` repository directly.

**Comment by dralley** [comment]
@omaciel Thanks, I was just assuming the status quo until that discussion was resolved.

**Comment by omaciel** [comment]
@dralley yup, totally understood and thank you for your understanding. @kersommoura and I will be presenting a new proposal soon with more details.

**Pull request closed without merging**